File: src/main/java/io/miret/etienne/hourglass/data/core/Event.java
package io.miret.etienne.hourglass.data.core;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.With;

import java.time.LocalDate;
import java.util.Set;
import java.util.UUID;

import static lombok.AccessLevel.PRIVATE;

@With
@Getter
@NoArgsConstructor (access = PRIVATE)
@AllArgsConstructor
public class Event {

  private UUID id;

  private String name;

  private LocalDate date;

  private UUID scaleRuleId;

  private Integer points;

  private Set<UUID> userIds;

}

File: src/main/ts/state/event.ts
import { HttpStatus } from './status';

export interface NewEvent {
  name: string;
  date: string;
  scaleRuleId: string;
  userIds: string[];
}

export interface Event extends NewEvent {
  id: string;
  points: number;
}

export interface EventContainer {
  edition: Event | null;
  event: Event;
  status: HttpStatus;
}
